frontend/static/js/pages/pongPage.js

```diff
@@ -1,31 +1,29 @@
-import ChatSocket from '../websocket.js';
-import { Pong } from '../components/pong.js';
+import ChatSocket from '../pong/ChatSocket.js';
+import PongService from '../services/pongService.js';
+import Pong from '../pong/pong.js';
 import AView from './AView.js';
+import PongGame from '../pong/pongGame.js';
+
 
 export default class extends AView {
 	constructor(params) {
 		super(params);
 		this.setTitle('Pong');
-	}
-
-	appendEventListeners() {
-		const chatSocket = ChatSocket();
-		document.addEventListener('keydown', (event) => {
-			const key = event.key;
-			document.getElementById('key-pressed').textContent = key;
-			chatSocket.send(
-				JSON.stringify({
-					message: key,
-				})
-			);
-		});
+		this.chatSocket = null;
+		this.pongService = new PongService();
 	}
 
 	async getHTML() {
+		let matchUrl = null;
+		try {
+			matchUrl = await this.pongService.getRequest();
+			this.chatSocket = new ChatSocket(matchUrl);
```

Comment:
Is it so that we stay in this page for the whole game, so we truly need to connect to a socket everytime one loads up this page? could one try to connect with another tab to the same socket etc?